lib/s3.js

```diff
@@ -0,0 +1,141 @@
+'use strict'
+
+const util = require('./util')
+const assert = require('assert')
+
+module.exports = makeClient
+
+function makeClient (options) {
+  let context = Object.assign({}, options)
+  context.logger = util.logWrapper(context.logger)
+  assert(context.region, 'Region is a required option for S3 clients')
+
+  return {
+    getObject: getObject.bind(null, context),
+    putObject: putObject.bind(null, context),
+    deleteObject: deleteObject.bind(null, context)
+  }
+}
+
+function getObject (context, params, callback) {
+  return util.optionalCallback(context, callback, util.signedRequest(context, {
+    method: 'GET',
+    path: `/${params.Bucket}/${encodeURIComponent(params.Key)}`
+  })
+  .then(result => {
+    let headers = result.headers
+    return {
+      Body: result.data,
+      DeleteMarker: headers['x-amz-delete-marker'],
+      AcceptRanges: headers['accept-ranges'],
+      Expiration: headers['x-amz-expiration'],
+      Restore: headers['x-amz-restore'],
+      LastModified: headers['Last-Modified'],
+      ContentLength: headers['Content-Length'],
+      ETag: headers['etag'],
+      MissingMeta: headers['x-amz-missing-meta'],
+      VersionId: headers['x-amz-version-id'],
+      CacheControl: headers['Cache-Control'],
+      ContentDisposition: headers['Content-Disposition'],
+      ContentEncoding: headers['Content-Encoding'],
+      ContentLanguage: headers['Content-Language'],
+      ContentRange: headers['Content-Range'],
+      ContentType: headers['Content-Type'],
+      Expires: headers['Expires'],
+      WebsiteRedirectLocation: headers['x-amz-website-redirect-location'],
+      ServerSideEncryption: headers['x-amz-server-side-encryption'],
+      Metadata: getMetadata(headers),
+      SSECustomerAlgorithm: headers['x-amz-server-side-encryption-customer-algorithm'],
+      SSECustomerKeyMD5: headers['x-amz-server-side-encryption-customer-key-MD5'],
+      SSEKMSKeyId: headers['x-amz-server-side-encryption-aws-kms-key-id'],
+      StorageClass: headers['x-amz-storage-class'],
+      RequestCharged: headers['x-amz-request-charged'],
+      ReplicationStatus: headers['"x-amz-replication-status'],
+      PartsCount: headers['x-amz-mp-parts-count'],
+      TagCount: headers['x-amz-tagging-count']
+    }
+  }))
+}
+
+function getMetadata (headers) {
+  if (!headers) return undefined
+  return Object.keys(headers)
+    .filter(h => h.indexOf('x-amz-meta-') === 0)
+    .reduce((obj, key) => {
+      obj[key] = headers[key]
+      return obj
+    }, {})
+}
+
+function deleteObject (context, params, callback) {
+  return util.optionalCallback(context, callback, util.signedRequest(context, {
+    method: 'DELETE',
+    path: `/${params.Bucket}/${encodeURIComponent(params.Key)}`
+  }).then(result => {
+    // context.logger.debug(result.headers, result.statusCode)
```

Comment:
If this was just for debugging we should remove it.  Otherwise, the point of log levels is to allow classes of messages to be enabled/disabled.